### DISCUSSION TITLE
Why is MAUI Essentials not a nuget package?

### DISCUSSION THREAD

**Discussion by jeanplevesque** [dotnet/maui#12867] [Q&A]
I'm curious to know why MAUI Essentials is enabled via a special property (`<UseMaui>true</UseMaui>`) instead of using a regular nuget package like Xamarin.Essentials?

**Edit**: I think there might be some issues with the special property.
- It's hard to know which version is being used.
- Can we even choose a version?
- It's unclear whether we can (or should) use this on non-MAUI projects (like WinUI 3, Uno Platform, etc.).
  - i.e. The supported target frameworks are unclear.

**Comment by symbiogenesis** [comment]
I agree with the implied preference. UseMaui should not hide package references. Makes everything more confusing.

But you could alleviate the issue by turning on nuget lock files and checking them into source control.

That way you would see all hidden package updates

https://www.meziantou.net/faster-and-safer-nuget-restore-using-source-mapping-and-lock-files.htm

**Comment by Dreamescaper** [comment]
@jeanplevesque 
If you don't need the whole Maui, but only essentials, you can use `<UseMauiEssentials>true</UseMauiEssentials>` instead.
However, this property controls not a regular PackageReference, but FrameworkReference, so the concern regarding versioning stands.

(I assume it will be easier once this PR is merged https://github.com/dotnet/maui/pull/11206)
